eng/packages/http-client-csharp/generator/Azure.Generator/src/Providers/MgmtLongRunningOperationProvider.cs

```diff
@@ -0,0 +1,226 @@
+                    _operationField.Invoke("WaitForCompletionResponse", [timeSpanParameter, cancellationTokenParameter]),
+                    this, XmlDocProvider.InheritDocs);
+            var waitForCompletionResponseAsyncMethod = _isGeneric
+                ? new MethodProvider(
+                    new MethodSignature("WaitForCompletionAsync", null, MethodSignatureModifiers.Public | MethodSignatureModifiers.Override, new CSharpType(typeof(ValueTask), new CSharpType(typeof(Response), _t)), null, [cancellationTokenParameter]),
+                    _operationField.Invoke("WaitForCompletionAsync", [cancellationTokenParameter]),
+                    this, XmlDocProvider.InheritDocs)
+                : new MethodProvider(
+                    new MethodSignature("WaitForCompletionResponseAsync", null, MethodSignatureModifiers.Public | MethodSignatureModifiers.Override, new CSharpType(typeof(ValueTask), typeof(Response)), null, [cancellationTokenParameter]),
+                    _operationField.Invoke("WaitForCompletionResponseAsync", [cancellationTokenParameter]),
+                    this, XmlDocProvider.InheritDocs);
+            var waitForCompletionResponseWithPollingAsyncMethod = _isGeneric
+                ? new MethodProvider(
+                    new MethodSignature("WaitForCompletionAsync", null, MethodSignatureModifiers.Public | MethodSignatureModifiers.Override, new CSharpType(typeof(ValueTask), new CSharpType(typeof(Response), _t)), null, [timeSpanParameter, cancellationTokenParameter]),
+                    _operationField.Invoke("WaitForCompletionAsync", [timeSpanParameter, cancellationTokenParameter]),
+                    this, XmlDocProvider.InheritDocs)
+                : new MethodProvider(
+                    new MethodSignature("WaitForCompletionResponseAsync", null, MethodSignatureModifiers.Public | MethodSignatureModifiers.Override, new CSharpType(typeof(ValueTask), typeof(Response)), null, [timeSpanParameter, cancellationTokenParameter]),
+                    _operationField.Invoke("WaitForCompletionResponseAsync", [timeSpanParameter, cancellationTokenParameter]),
+                    this, XmlDocProvider.InheritDocs);
+            return [BuildGetOperationIdMethod(), getRehydrationTokenMethod, getRawResponseMethod, updateStatusMethod, updateStatusAsyncMethod, waitForCompletionResponseMethod, waitForCompletionResponseWithPolingMethod, waitForCompletionResponseAsyncMethod, waitForCompletionResponseWithPollingAsyncMethod];
+        }
+
+        private const string GetOperationIdName = "GetOperationId";
+        private MethodProvider BuildGetOperationIdMethod()
+        {
+            var rehydrationTokenParaemter = new ParameterProvider("rehydrationToken", $"", new CSharpType(typeof(RehydrationToken), true));
```

Comment:
Why are we passing an empty description? Is this internal? Even still, a description might be helpful for customers.

Reply:
We did not have them in Autorest.CSharp, I agree it would be helpful, will add description.

Reply:
Added, but we are not generating docs for internal methods anyway.

Reply:
We should fix that in MGC. For properties we do generate internal docs.